src/configureStore.tsx

```diff
@@ -35,6 +37,53 @@ export type ConnectedReduxProps<A extends Action = AnyAction> = {
   dispatch: ThunkDispatch<A>;
 };
 
+const flattenObject = (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  object: Record<string, any>,
+) => {
+  return Object.keys(object).reduce((newObject: typeof object, k: string) => {
+    const value = object[k];
+    let newValue;
+
+    if (
+      value === null ||
+      ['string', 'number', 'boolean', 'undefined'].includes(typeof value)
+    ) {
+      newValue = value;
+    } else {
+      newValue = `[type: ${typeof value}]`;
+    }
+
+    // eslint-disable-next-line no-param-reassign
+    newObject[k] = newValue;
+    return newObject;
+  }, {});
+};
+
+export const actionToSentryBreadcrumb = (action: AnyAction) => {
+  return {
+    ...flattenObject(action),
+    payload: action.payload ? flattenObject(action.payload) : undefined,
+  };
+};
+
+export const redactStateForSentry = (state: ApplicationState) => {
+  // When adding a new state entry to this object, consider the
+  // implication of sending it to Sentry and redact data if necessary.
+  return {
+    accordionMenu: state.accordionMenu,
+    api: { authToken: '[redacted]' },
```

Comment:
Should we spread `state.api` before? (to show how to redact a state entry partially)

Reply:
No. I consciously didn't want to do that. Maybe I will add a comment saying not to do that.

Imagine that someone adds a *new* key to `api` called `api.authTokenSalt` or something. In this situation, I don't want the redacting code to accidentally leak the data. In my patch, the author of such a change will see [this test fail](https://github.com/mozilla/addons-code-manager/blob/390cefc345a74c30c48642859e8c662dc3f82b32/src/configureStore.spec.tsx#L84)  because it's checking that the redacted state at least has the same keys as the real state.

For practical reasons, I didn't choose to explicitly map each item within the other redacted reducers because that would create a maintenance burden. The maintenance burden on the `api` reducers seems reasonable since it will force the author to think about redaction as the `api` shape evolves.